ui/components/MesheryPerformance/PerformanceProfileTable.js

```diff
@@ -288,7 +288,8 @@ function MesheryTestProfiles({
         case "search":
           if (searchTimeout.current) clearTimeout(searchTimeout.current);
           searchTimeout.current = setTimeout(() => {
-            if (search !== tableState.searchText && tableState.searchText !== null) setSearch(tableState.searchText);
+            const regexp = /^\S/ //search should not start with space
```

Comment:
@piyushsingariya do we prevent performance tests from initially being named with a leading whitespace?

We might need to trim this on the endpoint handler side, if so.

Reply:
Yeah, we should, but it doesn't affect any current functionality, because we implement regex-based profile searching on the backend.

Reply:
And in Meshery UI?

Reply:
I think we can restrict people from adding leading whitespaces in Meshery-UI. It's a good issue to open.

Reply:
If there is trouble for leading whitespace, why we are wasting efforts in terms of regex checking and all. 
[Just use a trim for the search text. ](https://www.w3schools.com/jsref/jsref_trim_string.asp)

```diff
-            const regexp = /^\S/ //search should not start with space
- if (search !== tableState.searchText && regexp.test(tableState.searchText)) setSearch(tableState.searchText);
+  if (search !== tableState.searchText) setSearch(tableState.searchText.trim());
```

We can also live validate the text-box for whitespaces and all: (probably that won't be required, seeking the current use-case)
https://github.com/meshery/meshery/pull/4931

Reply:
Yes I think I'll just remove this regex and checking. If a user enters space he will be displayed with a `Sorry no matching profiles found` message. I guess that should be fine, as I checked this is the same behavior on the pattern tables too. @Abhishek-kumar09